<?xml version="1.0"?>
<launch>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 1.57 3.14 1.57 /world /map 100" />
    <node pkg="tf" type="static_transform_publisher" name="base2imu_tf"  args="0 0 0 0 0 0 /odom /camera_depth_optical_frame 100" />

    <include file="$(find r2dio)/launch/rs_camera.launch">
    </include>

    <param name="file_path" value="$(find r2dio)/config/params.yaml"/>

    <node pkg="r2dio" type="r2dio_laser_processing_node" name="r2dio_laser_processing_node" output="screen" />
    <node pkg="r2dio" type="r2dio_odom_estimation_node" name="r2dio_odom_estimation_node" output="screen" />
    <node pkg="r2dio" type="r2dio_laser_mapping_node" name="r2dio_laser_mapping_node" output="screen"/>

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find r2dio)/rviz/r2dio.rviz" />
    </group>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="r2dio" >
        <param name="/target_frame_name" value="map" />
        <param name="/source_frame_name" value="odom" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
